multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ProcessMtaArchiveStep.java

```diff
@@ -83,36 +95,58 @@ private void addMtaArchiveModulesInMtaArchiveElements(ProcessContext context, Mt
         mtaArchiveModules.forEach(mtaArchiveElements::addModuleFileName);
-        getStepLogger().debug("MTA Archive Modules: {0}", mtaArchiveModules.keySet());
+        getStepLogger().debug(Messages.MTA_ARCHIVE_MODULES_0_MESSAGE, mtaArchiveModules.keySet());
         context.setVariable(Variables.MTA_ARCHIVE_MODULES, mtaArchiveModules.keySet());
     }
 
     private void addMtaRequiredDependenciesInMtaArchiveElements(MtaArchiveHelper helper, MtaArchiveElements mtaArchiveElements) {
         Map<String, String> mtaArchiveRequiresDependencies = helper.getMtaRequiresDependencies();
         mtaArchiveRequiresDependencies.forEach(mtaArchiveElements::addRequiredDependencyFileName);
-        getStepLogger().debug("MTA Archive Requires: {0}", mtaArchiveRequiresDependencies.keySet());
+        getStepLogger().debug(Messages.MTA_ARCHIVE_REQUIRES_0_MESSAGE, mtaArchiveRequiresDependencies.keySet());
     }
 
     private void addMtaArchiveResourcesInMtaArchiveElements(MtaArchiveHelper helper, MtaArchiveElements mtaArchiveElements) {
         Map<String, String> mtaArchiveResources = helper.getMtaArchiveResources();
         mtaArchiveResources.forEach(mtaArchiveElements::addResourceFileName);
-        getStepLogger().debug("MTA Archive Resources: {0}", mtaArchiveResources.keySet());
+        getStepLogger().debug(Messages.MTA_ARCHIVE_RESOURCES_0_MESSAGE, mtaArchiveResources.keySet());
+    }
+
+    private DeploymentDescriptor extractDeploymentDescriptor(ProcessContext context, String appArchiveId,
+                                                             List<ArchiveEntryWithStreamPositions> archiveEntriesWithStreamPositions) {
+
+        ArchiveEntryWithStreamPositions deploymentDescriptorEntry = ArchiveEntryExtractorUtil.findEntry(ArchiveHandler.MTA_DEPLOYMENT_DESCRIPTOR_NAME,
+                                                                                                        archiveEntriesWithStreamPositions);
+        byte[] inflatedDeploymentDescriptor = archiveEntryExtractor.readFullEntry(ImmutableFileEntryProperties.builder()
+                                                                                                              .guid(appArchiveId)
+                                                                                                              .name(deploymentDescriptorEntry.getName())
+                                                                                                              .spaceGuid(context.getRequiredVariable(Variables.SPACE_GUID))
+                                                                                                              .maxFileSizeInBytes(configuration.getMaxMtaDescriptorSize())
+                                                                                                              .build(),
+                                                                                  deploymentDescriptorEntry);
```

Comment:
Can you make this more generic and use also in the createMtaArchiveHelperFromManifest method